src/Orckestra.Composer.Cart/Services/CartService.cs

```diff
@@ -117,7 +116,9 @@ public async virtual Task<CartViewModel> AddLineItemAsync(AddLineItemParam param
 
             try
             {
-                cart = await CartRepository.AddLineItemAsync(param).ConfigureAwait(false);
+                cart = EditingOrderProvider.IsEditMode()
```

Comment:
I think we can use here param.CartType = OrderDraft
also can we have something like TryGetExisingLineItem(,,,, out existingLineItem

and the condition 
if(param.CartType = OrderDraft && TryGetExisingLineItem(,,,, out existingLineItem ) {
/// execute update
} else {
execute add 
}